crates/uv-workspace/src/pyproject_mut.rs

```diff
@@ -84,6 +87,160 @@ impl ArrayEdit {
     }
 }
 
+/// The default version specifier when adding a dependency.
+// While PEP 440 allows an arbitrary number of version digits, the `major` and `minor` build on
+// most projects sticking to two or three components and a SemVer-ish versioning system, so can
+// bump the major or minor version of a major.minor or major.minor.patch input version.
+#[derive(Clone, Copy, Debug, Default, Deserialize, PartialEq, Eq, Serialize)]
+#[serde(rename_all = "kebab-case")]
+#[cfg_attr(feature = "clap", derive(clap::ValueEnum))]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+pub enum AddBoundsKind {
+    /// Only a lower bound, e.g., `>=1.2.3`.
+    #[default]
+    Lower,
+    /// Allow the same major version, similar to the semver caret, e.g., `>=1.2.3,<2.0.0`.
+    ///
+    /// Leading zeroes are skipped, e.g. `>=0.1.2,<0.2.0`.
+    Major,
+    /// Allow the same minor version, similar to the semver tilde, e.g., `>=1.2.3,<1.3.0`.
+    ///
+    /// Leading zeroes are skipped, e.g. `>=0.1.2,<0.1.3`.
+    Minor,
+    /// Pin the exact version, e.g., `==1.2.3`.
+    ///
+    /// This option is not recommended, as uv provides a lockfiles that pins versions.
+    Exact,
+}
+
+impl AddBoundsKind {
+    fn specifiers(self, version: Version) -> VersionSpecifiers {
```

Comment:
Do I need to review this closely? Should we leverage another reviewer like @BurntSushi so I can focus on the ux?